### PR TITLE
[onert] Update TrainingInfo::isValid() to return error mesage

### PR DESCRIPTION
**Pull request by zetwhite**
This PR is to update TrainingInfo::isValid() to return error message.

ONE-DCO-1.0-Signed-off-by: SeungHui Youn <sseung.youn@samsung.com>

### CODE REVIEW

File: runtime/onert/api/src/nnfw_api_internal.cc [hunk 1]
```diff
@@ -1346,8 +1346,9 @@ NNFW_STATUS nnfw_session::train_prepare()
 
   try
   {
-    if (not _train_info->isValid())
-      throw std::runtime_error{"training info is not valid"};
+    const auto res = _train_info->isValid();
+    if (not res.valid)
+      throw std::runtime_error{"training info is not valid \n" + res.error_msg};
```
Comment (zetwhite) [inline runtime/onert/api/src/nnfw_api_internal.cc:1351]: After this PR, `train_prepare` show that - why training info is invalid. 

### Before this PR 

When the user sets invalid training info, 
`train_prepare`fails without showing which training information is invalid.

```
./onert_train --modelfile mnist_with_meta.circle --load_input:raw pool_mnist_model/mnist_x_train.1000.bin --load_expected:raw pool_mnist_model/mnist_y_train.1000.bin --epoch 10  --learning_rate=0 --batch_size=0

Model Expected Filename pool_mnist_model/mnist_y_train.1000.bin
Model Input Filename pool_mnist_model/mnist_x_train.1000.bin
Model Filename mnist_with_meta.circle
== training parameter ==
- learning_rate   = 0
- batch_size      = 0
- loss_info       = {loss = mean squared error, reduction = use automatic reduction type}
- optimizer       = sgd
========================
Error during nnfw_session::train_prepare : training info is not valid 
```

### After this PR 

`train_prepare` shows the message like 'batch size must not be zero', 'learning rate must be positive'. 

```bash 
./onert_train --modelfile mnist_with_meta.circle --load_input:raw pool_mnist_model/mnist_x_train.1000.bin --load_expected:raw pool_mnist_model/mnist_y_train.1000.bin --epoch 10  --learning_rate=0 --batch_size=0

Model Expected Filename pool_mnist_model/mnist_y_train.1000.bin
Model Input Filename pool_mnist_model/mnist_x_train.1000.bin
Model Filename mnist_with_meta.circle
== training parameter ==
- learning_rate   = 0
- batch_size      = 0
- loss_info       = {loss = mean squared error, reduction = use automatic reduction type}
- optimizer       = sgd
========================
Error during nnfw_session::train_prepare : training info is not valid 
batch size must not be zero
learning rate must be positive

```

**Comment by jyoungyun** [comment]
I don't want to interrupt you, but I'm not sure if this PR is really necessary. I think it would be better to give a well-described document than to print information of all parameters as error messages.

File: runtime/onert/api/src/nnfw_api_internal.cc [hunk 2]
```diff
@@ -1346,8 +1346,9 @@ NNFW_STATUS nnfw_session::train_prepare()
 
   try
   {
-    if (not _train_info->isValid())
-      throw std::runtime_error{"training info is not valid"};
+    const auto res = _train_info->isValid();
+    if (not res.valid)
```
Comment (glistening) [inline runtime/onert/api/src/nnfw_api_internal.cc:1350]: Just a note. Runtime have not used `not`, we've used `!` so far. I am okay for using alternative tokens.

**Comment by zetwhite** [comment]
> I don't want to interrupt you, but I'm not sure if this PR is really necessary. 

Thanks for the opinion :) 
As you pointed out, this isn't not the one which is really necessary.
My intention was providing a kind API - pointing out detailed reason why it fails to compile(prepare).
I'm also satisfied with giving detailed description, instead of giving detailed error message.

**Comment by zetwhite** [comment]
Let's close this one for now.
I'll reopen this when it become really necessary.
